### ISSUE TITLE
Fix AzDO (read-only) variables in Arcade

### ISSUE THREAD

**Issue by missymessa** [dotnet/arcade#5422]
Related to https://github.com/dotnet/arcade/issues/5085

The following is a list of all AzDO variables in Arcade that we need to determine if they need to be refactored to be made not read-only: 
- BARBuildId https://github.com/dotnet/arcade/blob/18ac1be0e3520e91b7a2332fe98dd4a8ebe47dbe/eng/common/templates/post-build/setup-maestro-vars.yml#L62
- TargetChannels https://github.com/dotnet/arcade/blob/18ac1be0e3520e91b7a2332fe98dd4a8ebe47dbe/eng/common/templates/post-build/setup-maestro-vars.yml#L63
- Helix_JobToken https://github.com/dotnet/arcade/blob/b58add62b88ab28edd15932fb19f00b9e1c5e8dd/eng/common/templates/steps/telemetry-start.yml#L83
- Helix_WorkItemId https://github.com/dotnet/arcade/blob/b58add62b88ab28edd15932fb19f00b9e1c5e8dd/eng/common/templates/steps/telemetry-start.yml#L138
- AzDORepoNameName https://github.com/dotnet/arcade/blob/27b56f866fe9a97cdc02dc439082476dae6fd235/azure-pipelines-merge-mirror.yml#L34
- TargetBranchName https://github.com/dotnet/arcade/blob/27b56f866fe9a97cdc02dc439082476dae6fd235/azure-pipelines-merge-mirror.yml#L35
- VSS_NUGET_EXTERNAL_FEED_ENDPOINTS https://github.com/dotnet/arcade/blob/a68b528ad1d0305f55c4fbdc977ebb0c02cfcc19/eng/common/internal-feed-operations.sh#L68
- NUGET_CREDENTIALPROVIDER_SESSIONTOKENCACHE_ENABLED https://github.com/dotnet/arcade/blob/a68b528ad1d0305f55c4fbdc977ebb0c02cfcc19/eng/common/internal-feed-operations.sh#L69
- AzDORepoName https://github.com/dotnet/arcade/blob/879c48df89a111a03c4b69b830588e7a15b1c5be/azure-pipelines-code-mirror.yml#L32

Here are some dynamically set variables that we'll also need to handle: 
- internal-feed-operations.ps1 https://github.com/dotnet/arcade/blob/a68b528ad1d0305f55c4fbdc977ebb0c02cfcc19/eng/common/internal-feed-operations.ps1#L70
https://github.com/dotnet/arcade/blob/a68b528ad1d0305f55c4fbdc977ebb0c02cfcc19/eng/common/internal-feed-operations.ps1#L76
- pipeline-logging-functions.sh https://github.com/dotnet/arcade/blob/a68b528ad1d0305f55c4fbdc977ebb0c02cfcc19/eng/common/pipeline-logging-functions.sh#L151

After handling these, we'll need to document guidance for working with AzDO variables in the future.

**Comment by missymessa** [comment]
Private Arcade build with all the AzDO variables set to read-only: https://dnceng.visualstudio.com/internal/_build/results?buildId=635781&view=results

**Comment by missymessa** [comment]
Build of Arcade Validation using the above private build of Arcade: https://dnceng.visualstudio.com/internal/_build/results?buildId=636848&view=results

**Comment by missymessa** [comment]
I changed all the AzDO variables in a private branch of Arcade to be read-only, and didn't see any errors when I ran that build through Arcade Validation. In the parent issue, I checked with Matt Cooper to see what kind of messaging I would see if the feature flag was turned on, and it looks like the official Arcade build (https://dnceng.visualstudio.com/internal/_build?definitionId=6&_a=summary) doesn't have the warning that we would expect if there were read-only variables being overwritten. Closing this issue.
